### PR TITLE
feat(source-map): add support for compiled test run

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/testModel.ts [hunk 1]
```diff
@@ -168,11 +186,12 @@ export class TestModel {
   }
 
   async listTests(files: string[]) {
-    const filesToLoad = files.filter(f => this.allFiles.has(f));
-    if (!filesToLoad.length)
+    const sourcestoLoad = files.filter(f => this.allFiles.has(f));
```
Comment (mxschmitt) [inline src/testModel.ts:189]:
```suggestion
    const sourcesToLoad = files.filter(f => this.allFiles.has(f));
```

File: src/utils.ts [hunk 2]
```diff
@@ -150,3 +151,37 @@ export async function spawnAsync(executable: string, args: string[], cwd?: strin
     childProcess.on('exit', () => f(output));
   });
 }
+
+export async function resolveSourceMap(file: string, fileToSources: Map<string, string[]>, sourceToFile: Map<string, string>): Promise<string[]> {
+  if (!file.endsWith('.js'))
```
Comment (mxschmitt) [inline src/utils.ts:156]: mjs or cjs? probably not something where rollup or tsc compile to I guess?

File: src/utils.ts [hunk 3]
```diff
@@ -150,3 +151,37 @@ export async function spawnAsync(executable: string, args: string[], cwd?: strin
...
+
+  const rl = readline.createInterface({ input: fs.createReadStream(file), crlfDelay: Infinity });
+
+  let lastLine: string | undefined;
+  rl.on('line', line => {
+    lastLine = line;
+  });
+  await new Promise(f => rl.on('close', f));
+
+  if (lastLine?.startsWith('//# sourceMappingURL=')) {
```
Comment (mxschmitt) [inline src/utils.ts:170]: what about inline sourcemaps? Not sure if they make sense here.

**Review by mxschmitt — APPROVED** [review]

File: src/utils.ts (hunk 3 above)
Comment (pavelfeldman) [reply src/utils.ts:170]: We might need to support them at some point, but they are not the tsc/esbuild defaults.

**Pull request merged**